src/csg/constraints.jl

```diff
@@ -0,0 +1,31 @@
+abstract type Constraint end
```

Comment:
Should we immediately move the implementation of constraints to a separate `Constraints.jl` package?
That is, we can define an abstract type `Constraint` here and move all constraints and their propagators to a different package. 

Then Constraints.jl will depend on Grammar.jl, and Search.jl uses then both Grammar.jl and Constraints.jl.

This can be done later also, but just ot raise a discussion about that.

Reply:
I think it is a very good idea to do it now, as separating might be more difficult later.

Reply:
I separated them. Therefore, there are now four related pull requests, which I all linked to this issue: https://github.com/Herb-AI/Herb.jl/issues/25